### PR TITLE
docs: Add Scala 3.6.0 postmortem blog post

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

File: _posts/2024-10-24-post-mortem-3.6.0.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+by: Scala Core Team
+---
+
+Incident Date: _October 18th, 2024_  
+Nature of the incident: _Accidental release of 3.6.0-RC1 under 3.6.0 on Maven
+Central_
+
+On Friday, 18th of October 2024, during the publication process of **3.6.0-RC1**
+an incident occurred that ended up in publishing Scala **3.6.0** instead. After
+an internal investigation, we pinpointed the incident to be of technical nature.
```
Comment (SethTisue) [inline _posts/2024-10-24-post-mortem-3.6.0.md:14]:
```suggestion
an internal investigation, we pinpointed the incident to be technical in nature.
```

File: _posts/2024-10-24-post-mortem-3.6.0.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+Within one hour of catching the issue, the
```
Comment (SethTisue) [inline _posts/2024-10-24-post-mortem-3.6.0.md:16]:
```suggestion
Within one hour of the issue being reported, the
```

File: _posts/2024-10-24-post-mortem-3.6.0.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+[Scala Core Team](https://scala-lang.org/scala-core/) mitigated the problem and
```
Comment (SethTisue) [inline _posts/2024-10-24-post-mortem-3.6.0.md:17]:
```suggestion
[Scala Core Team](https://scala-lang.org/scala-core/) mitigated the problem by
```

File: _posts/2024-10-24-post-mortem-3.6.0.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+## What happened?
+
+During the publication of Scala **3.6.0-RC1** to the Sonatype Maven
+repository**,** a mistake occurred and we released **a misconfigured 3.6.0**
+instead. As a result, the released version of the compiler produces TASTy files
+in the experimental version that cannot be consumed by stable TASTy consumers
+(e.g. the Scala compiler, tasty-query, Scaladoc). Setting the TASTy version to
+an experimental one is a standard procedure when releasing a new release
+candidate of a minor version of Scala 3\.
```
Comment (Philippus) [inline _posts/2024-10-24-post-mortem-3.6.0.md:59]:
```suggestion
candidate of a minor version of Scala 3.
```

File: _posts/2024-10-24-post-mortem-3.6.0.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+Nature of the incident: _Accidental release of 3.6.0-RC1 under 3.6.0 on Maven
+Central_
+
+On Friday, 18th of October 2024, during the publication process of **3.6.0-RC1**
+an incident occurred that ended up in publishing Scala **3.6.0** instead. After
+an internal investigation, we pinpointed the incident to be of technical nature.
+
+Within one hour of catching the issue, the
+[Scala Core Team](https://scala-lang.org/scala-core/) mitigated the problem and
+started the Scala **3.6.1** release – that is to be treated as RC, which is
```
Comment (SethTisue) [inline _posts/2024-10-24-post-mortem-3.6.0.md:18]:
```suggestion
starting the Scala **3.6.1** release – which is to be treated like an RC; it has been
```

File: _posts/2024-10-24-post-mortem-3.6.0.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+Use version 3.6.1 as an RC version, to be used only for testing.  
+DO NOT USE 3.6.0, not for testing, not for anything, it’s considered broken.
+
+We are working on 3.6.2 as the new stable release, which will be available
+sometime in the second half of November.
+
+## How does it affect users?
+
+Scala 3.6.0 is a broken release \- it should never be used by users of Scala.
+The results of the compilation might not be consumed by stable versions of the
```
Comment (SethTisue) [inline _posts/2024-10-24-post-mortem-3.6.0.md:43]:
```suggestion
The results of the compilation might not be consumable by stable versions of the
```

File: _posts/2024-10-24-post-mortem-3.6.0.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+compiler or TASTy based tooling.  
+Tooling that automatically uses the latest available version of Scala will use
+the hot-fix release 3.6.1, which should be treated as a Release Candidate
+version. We don’t encourage most users to upgrade yet, unless for testing
+purposes. Instead, use the stable 3.5.2 and wait with upgrades until 3.6.2 is
```
Comment (SethTisue) [inline _posts/2024-10-24-post-mortem-3.6.0.md:48]:
```suggestion
purposes. Instead, use the stable 3.5.2 and wait to upgrade until 3.6.2 is
```

File: _posts/2024-10-24-post-mortem-3.6.0.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+compiler to produce artifacts with the experimental version of TASTy. In
+particular, we wanted to prevent publishing Scala libraries from being published
+to the (immutable) Maven repository using a broken Scala release. It could also
+potentially affect applications using Scala 3.6.0 that would try to use
+libraries produced by Scala 3.6.1 or later.
+
+We decided to publish Scala 3.6.1 as a follow-up ASAP so that the most recent
+version doesn’t include the experimental TASTy flag nor RC settings. With 3.6.1
+we ensured tooling will, at the very least, refer to a Scala version that
+actually could be considered stable.
```
Comment (SethTisue) [inline _posts/2024-10-24-post-mortem-3.6.0.md:93]:
```suggestion
actually could be considered reasonably stable.
```

File: _posts/2024-10-24-post-mortem-3.6.0.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+the following processes as preventative measures:
+
+- Extra checks to the release procedure to ensure the correct versioning of
+  published Scala artifacts.
+- Any changes related to either release scripts or compiler versioning will be
+  tested using a custom Maven repository before the start of a public release.
+- Additional, manual approval of published artifacts to the staging Maven
+  repository.
+- We will be mindful of the day of week for releases, and avoid Thursdays and
+  Fridays at all costs.
```
Comment (SethTisue) [inline _posts/2024-10-24-post-mortem-3.6.0.md:138]:
```suggestion
  Fridays.
```

File: _posts/2024-10-24-post-mortem-3.6.0.md [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+Tooling that automatically uses the latest available version of Scala will use
+the hot-fix release 3.6.1, which should be treated as a Release Candidate
+version. We don’t encourage most users to upgrade yet, unless for testing
+purposes. Instead, use the stable 3.5.2 and wait with upgrades until 3.6.2 is
+out.
+
+## What happened?
+
+During the publication of Scala **3.6.0-RC1** to the Sonatype Maven
+repository**,** a mistake occurred and we released **a misconfigured 3.6.0**
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply _posts/2024-10-24-post-mortem-3.6.0.md:54]: Ach, shouldn't have trusted google docs to export correctly. Will check the original doc, thanks!

[the remark this answers is not in the document]
Comment (tgodzik) [reply _posts/2024-10-24-post-mortem-3.6.0.md:54]: Actually, the coma was bold in the doc, so google docs did the correct thing. Will fix, thanks again!

File: _posts/2024-10-24-post-mortem-3.6.0.md (hunk 4 above)
Comment (tgodzik) [reply _posts/2024-10-24-post-mortem-3.6.0.md]: Thanks, fixed!

**Review by WojciechMazur — APPROVED** [review]

**Pull request merged**